src/highdicom/seg/sop.py

```diff
@@ -1036,8 +1124,8 @@ def _check_and_cast_pixel_array(
     @staticmethod
     def _omit_empty_frames(
         pixel_array: np.ndarray,
-        plane_positions: Sequence[PlanePositionSequence]
-    ) -> Tuple[np.ndarray, Sequence[PlanePositionSequence], List[int]]:
+        plane_positions: Sequence[Optional[PlanePositionSequence]]
+    ) -> Tuple[np.ndarray, List[Optional[PlanePositionSequence]], List[int]]:
```

Comment:
Should maybe the entire list be optional? `Optional[Sequence[PlanePositionSequence]]`

Reply:
I agree that that would seem more obvious on the face of it, especially if it were part of the public API. But actually I think it would result in quite considerably more complex branching logic both inside this method and in the surrounding code that calls it, e.g. zipped/enumerated for loops using this list. The way things are now, the same processes can continue with a list of plane position items without having to worry about whether or not those items are meaningful, and then only worry at the end about whether or not to actually include the plane positions in the dataset.

Either way is a bit ugly, but I think I favour this way